### DISCUSSION TITLE
Upgrade/Migrate from keycloak 15 (wildfly) to 17 (quarkus) incomplete

### DISCUSSION THREAD

**Discussion by stekaa** [keycloak/keycloak#11169] [Archive]
I just upgraded from KC 15 to 17. 
Database-Connection to MariaDB seams to work because I can login in adminconsole with the old credentials.

After logging in there is only the top bar with my adminuser top right. No content or even left menu bar exists. There are only the default links in user-icon, but they doesn't work:

1. "Server Info": does/shows nothing

2. "Sign Out": ok this one works ;-) 

3. "Manage account": gives "We are sorry... Invalid parameter: redirect_uri"

  In console is only one entry: 

  `WARN  [org.keycloak.events] (executor-thread-2) type=LOGIN_ERROR, realmId=master, clientId=account-console, userId=null, ipAddress=<MyClientIP>, error=invalid_redirect_uri, redirect_uri=https://<keycloak-domain>/realms/master/account/?referrer=security-admin-console&referrer_uri=https%3A%2F%2F<keycloak-domain>%2Fadmin%2Fmaster%2Fconsole%2F%23%2Fserver-info`

Do You have any hints regarding 
- redirect_uri
- blank page after logging in

side nodes: Keycloak runs behind apache reverse proxy

**Comment by DGuhr** [comment]
Please make sure you have set the hostname to a valid one (e.g. no more than 253 chars, in all-lowercase....) At least that fixed it for #10930 - i guess that may be your problem. Also there is https://github.com/keycloak/keycloak/issues/11083 where i provided a fix for using hostname and standard https port without a proxy. You could try out the latest nightly build of main to see if the error is gone with that. (See https://quay.io/repository/keycloak/keycloak?tab=tags )  

Thats all i can say without having more details. Hope it helps :)

**Reply by stekaa** [reply]
Thanks a lot for Your answer. It saved additional hours of investigating.
The reason for my problems were capital letters in hostname. I just used them for better readability.

I think it should be written in documentation (e.g. https://www.keycloak.org/server/all-config): **hostname must be lower case**.

**Reply by DGuhr** [reply]
it is, and will be there with the next keycloak version - see https://github.com/keycloak/keycloak/blob/main/docs/guides/src/main/server/hostname.adoc#hostname-syntax unfortunately atm the docs release is tied to a keycloak release.
